cli/src/maintenance.rs

```diff
@@ -439,29 +439,34 @@ impl SolidoState {
 
     /// If there is a validator being deactivated, try to unstake its funds.
     pub fn try_unstake_from_inactive_validator(&self) -> Option<(Instruction, MaintenanceOutput)> {
-        for validator in &self.solido.validators.entries {
-            // We are only interested in unstaking from inactive validators.
-            if validator.entry.active {
+        for (validator, stake_accounts) in self
+            .solido
+            .validators
+            .entries
+            .iter()
+            .zip(self.validator_stake_accounts.iter())
+        {
+            // We are only interested in unstaking from inactive validators that
+            // have stake accounts.
+            if validator.entry.active || stake_accounts.is_empty() {
+                continue;
+            }
+            // Validator already has 3 unstake accounts.
+            if validator.entry.unstake_seeds.end - validator.entry.unstake_seeds.begin >= 3 {
```

Comment:
We should make this 3 a constant (maybe in `program/src/lib.rs`), instead of repeating it in multiple places.